apps/server/src/controllers/sessionContorller.js

```diff
@@ -17,16 +17,38 @@ module.exports = {
     }
   },
   async updateSession(req, res) {
+    // need to create a new bankroll entry when a session is completed
+    // get most recent bankroll entry
+    // figure out how to access the PLUS/MINUS on the session
+    // create new bankroll with previous bankroll + PLUS/MINUS
     try {
       const session = await Session.findOneAndUpdate(
         { _id: new mongoose.Types.ObjectId(req.body.id) },
         { $set: { ...req.body, endTime: Date.now() } },
         { new: true }
       );
+      console.log(session);
       if (!session) {
         return res.status(404).json({ message: 'No session with this id!' });
       }
-      res.json('Session Updated Sucessfully');
+      let bankroll = await Bankroll.findOne(
+        // find most recent bankroll with the userID
+        { _id: new mongoose.Types.ObjectId(req.body.id) },
+        {},
+        { sort: { timeStamp: -1 } }
+      );
```

Comment:
Yeah I definitely misled you here with `_id`, that's Mongo's identifier but you want to match on the user id instead. Check how you're finding the bankroll in your `bankrollController` methods.